vision_datasets/common/balanced_instance_weights_generator.py

```diff
@@ -0,0 +1,63 @@
+"""Instance weights generator for DataManifest. Only works for classification, detection, multitask. Adapted from irispy:
+https://msazure.visualstudio.com/Cognitive%20Services/_git/API-CustomVision-Irispy?path=/irispy/datasets/balanced_instance_weights_generator.py&version=GBmaster&_a=contents
+"""
+
+import logging
+from collections import Counter
+
+import numpy
+
+logger = logging.getLogger(__name__)
+
+
+class BalancedInstanceWeightsGenerator(object):
+    NEG_CLASS_INDEX = -1
+
+    @staticmethod
+    def generate(data_manifest, soft=True, weight_upper=5, weight_lower=0.2):
+        assert data_manifest
+
+        logging.info("Generating instance weights for dataset balancing.")
+        image_tags = [BalancedInstanceWeightsGenerator._process_tags(x.labels) for x in data_manifest.images]
+
+        class_wise_image_counter = Counter()
+        for tags in image_tags:
+            class_wise_image_counter.update(tags)
+
+        mean_class_wise_image_tag_count = numpy.mean(list(class_wise_image_counter.values()))
+        class_wise_multipliers = {x: mean_class_wise_image_tag_count / class_wise_image_counter[x] for x in class_wise_image_counter}
```

Comment:
no risk of zero division?

Reply:
I guess it's ok. But it's good to have a test case with a missing class if we didn't have

Reply:
I don't think so. Since `class_wise_image_counter` is a Counter, every key will have positive value.